tests/api_test.py

```diff
@@ -46,6 +46,9 @@
 
 class APITest(jtu.JaxTestCase):
 
+  def tearDown(self) -> None:
```

Comment:
Why not put this in jtu.JaxTestCase?

Reply:
I thought of that! But then I thought perhaps for some cases it would add nontrivial overhead, like where we have a zillion small test cases in lax_test.py or numpy_test.py (which also couldn't plausibly affect the tracer state).

Perhaps that was premature optimization, though... WDYT?

Reply:
After some follow-up, I'm going to try it!

Reply:
Following up in #2592.